### DISCUSSION TITLE
Feature: Add URL parameters for view configuration in explore

### DISCUSSION THREAD

**Discussion by Mario-Hofstaetter** [grafana/grafana#48633] [Explore]
When using structured logging, a VERY helpful feature in logs is to only show specific detected fields, e.g. `message` , instead of the complete JSON line in explore.

However when sharing the URL to this explore view, those selections are not included as URL query parameters.
When the recipent opens the link, he will see the whole JSON logs and has to again toggle `Show this field instead of the message`.

Using the "`Copy shortened link`" feature in explore does not change the behavior.

To be clear this is the feature mentioned: <https://grafana.com/docs/grafana/latest/explore/logs-integration/#toggle-detected-fields>

Additionally, the same is true for these toggles:

- `Time`
- `Unique labels`
- `Wrap lines`
- `Wrap lines`
- `Prettify JSON`
- `Dedup`
- `Display results` (Order)

It appears those are only saved in the browser session / reset to default when opening a new private window?

## Feature Request

Add configuration view options to URL as parameters so the view can be "reproduced" exactly as currently seen.

## Example

<details>
  <summary>This View (Click to expand!)</summary>
  
![grafik](https://user-images.githubusercontent.com/33002073/166453621-5ba290b9-128f-4f68-9761-75566ee32656.png)
  
</details>

Has this URL:
<https://play.grafana.org/explore?orgId=1&left=%7B%22datasource%22:%22Loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bfilename%3D%5C%22%2Fvar%2Flog%2Fnginx%2Fjson_access.log%5C%22%7D%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D>

Readable:
```text
https://play.grafana.org/explore?orgId=1&left={"datasource":"Loki","queries":[{"refId":"A","expr":"{filename=\"/var/log/nginx/json_access.log\"}"}],"range":{"from":"now-1h","to":"now"}}
```

Opening this link in a private browser window will reset the settings:

<details>
  <summary>(Click to expand!)</summary>
  
![grafik](https://user-images.githubusercontent.com/33002073/166454073-6092a715-f3bd-4e46-ad05-fdf1737777ea.png)
  
</details>

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
